File: src/main/java/net/shadowfacts/projectliftoff/api/internal/InternalMethods.java
package net.shadowfacts.projectliftoff.api.internal;

import net.shadowfacts.projectliftoff.api.dimension.DimensionInfoRegistry;

/**
 * @author shadowfacts
 */
public interface InternalMethods {

	InternalMethods NO_OP = new InternalMethods() {};

	default DimensionInfoRegistry getDimensionInfoRegistry() {
		return DimensionInfoRegistry.NO_OP;
	}

}

File: src/main/kotlin/net/shadowfacts/projectliftoff/internal/InternalMethods.kt
package net.shadowfacts.projectliftoff.internal

import net.shadowfacts.projectliftoff.api.dimension.DimensionInfoRegistry
import net.shadowfacts.projectliftoff.api.internal.InternalMethods
import net.shadowfacts.projectliftoff.dimension.DimensionRegistry

/**
 * @author shadowfacts
 */
object InternalMethods : InternalMethods {

	override fun getDimensionInfoRegistry(): DimensionInfoRegistry {
		return DimensionRegistry
	}

}